src/spec-lang/tc/typeenv.ts

```diff
@@ -91,8 +66,8 @@ export class TypeEnv {
     constructor(compilerVersion: string, abiEncoderVersion: ABIEncoderVersion) {
         this.typeMap = new Map();
 
-        this.userFunctions = new UserFunctionScoping();
-        this.userConstants = new UserConstantScoping();
+        this.userFunctions = new UserDefinitionScoping<SUserFunctionDefinition>();
+        this.userConstants = new UserDefinitionScoping<SUserConstantDefinition>();
```

Comment:
Should these be `UserFunctionScoping` and `UserConstantScoping` instead? We have declared class type aliases but not using them, otherwise.

Reply:
Whenever I try to use those instead I get a "type used as value" error.